### 1) Известно, что генеральная совокупность распределена нормально со средним квадратическим отклонением, равным 16.
### Найти доверительный интервал для оценки математического ожидания a с надежностью 0.95, если выборочная средняя M = 80, а объем выборки n = 256.

![условие](Задача_6_1.jpg)

In [10]:
left=80-1.96*16/256**(1/2)
right=80+1.96*16/256**(1/2)
print(f'95%-й доверительный интервал для оценки мат. ожидания генеральной совокупности: [{left};{right}].')

95%-й доверительный интервал для оценки мат. ожидания генеральной совокупности: [78.04;81.96].


#### Ответ:
    
**[78.04;81.96]**

### 2) В результате 10 независимых измерений некоторой величины X, выполненных с одинаковой точностью, получены опытные данные: 6.9, 6.1, 6.2, 6.8, 7.5, 6.3, 6.4, 6.9, 6.7, 6.1 
### Предполагая, что результаты измерений подчинены нормальному закону распределения вероятностей, оценить истинное значение величины X при помощи доверительного интервала, покрывающего это значение с доверительной вероятностью 0,95.

![условие](Задача_6_2.jpg)

In [11]:
import numpy as np
import scipy.stats as stats

In [45]:
arr=np.array([6.9, 6.1, 6.2, 6.8, 7.5, 6.3, 6.4, 6.9, 6.7, 6.1])
print(f'Среднее выборочное: {np.mean(arr): .2f},\n'
      f'Размер выборки n={len(arr)},\n'
      f'Среднее квадратическое отклонение по выборке(несмещенное): {np.std(arr, ddof=1): .2f}.')

Среднее выборочное:  6.59,
Размер выборки n=10,
Среднее квадратическое отклонение по выборке(несмещенное):  0.45.


In [13]:
def t_from_table(confidens, len_array):
    alpha=(1-confidens)
    return stats.t.ppf(1-alpha/2, len_array-1)
print(f'Табличное значение t-критерия для 95%-го доверительного интервала данной выборки: {t_from_table(0.95, len(arr)): .3f}')

Табличное значение t-критерия для 95%-го доверительного интервала данной выборки:  2.262


In [14]:
def confidens_int(arr, confidens):
    return round(np.mean(arr)-t_from_table(confidens,len(arr))*np.std(arr, ddof=1)/len(arr)**0.5,3), \
           round(np.mean(arr)+t_from_table(confidens,len(arr))*np.std(arr, ddof=1)/len(arr)**0.5,3)

print(f'95%-й доверительный интервал для истинного значения Х: {confidens_int(arr, 0.95)}.')

95%-й доверительный интервал для истинного значения Х: (6.268, 6.912).


#### Ответ:
    
**[6.268;6.912]**

### 3) Рост дочерей 175, 167, 154, 174, 178, 148, 160, 167, 169, 170
### Рост матерей 178, 165, 165, 173, 168, 155, 160, 164, 178, 175
### Используя эти данные построить 95% доверительный интервал для разности среднего роста родителей и детей.

In [33]:
import math
import scipy.stats as stats

In [34]:
mother = np.array([172, 177, 158, 170, 178,175, 164, 160, 169, 165])
daughter = np.array([173, 175, 162, 174, 175, 168, 155, 170, 160, 163])
n = len(mother)
alpha = 0.05
m1 = np.mean(mother)
m2 = np.mean(daughter)
delta = abs(m1 - m2)

In [44]:
print(f'Разность средних = {delta: .2f}')

Разность средних =  1.30


In [46]:
t = stats.t.ppf(1 - (alpha / 2), 2 * (n - 1))
print(f'Табличное значение t-критерия = {t: .3f}')

Табличное значение t-критерия =  2.101


In [47]:
d = (np.var(mother, ddof=1) + np.var(daughter, ddof=1)) / 2
print(f'Дисперсия двух групп = {d: .3f}')

Дисперсия двух групп =  49.339


In [53]:
s = math.sqrt((d / n) + (d / n))
print(f'Sdelta = {s: .3f}')

Sdelta =  3.141


In [41]:
result1 = delta - t * s
result2 = delta + t * s

In [55]:
print(
    f'Доверительный интервал для разности среднего роста родителей и детей с доверительной вероятностью 0,95 = ({result1: .3f},{result2: .3f})')

Доверительный интервал для разности среднего роста родителей и детей с доверительной вероятностью 0,95 = (-5.300, 7.900)


#### Ответ:
    
**[-5.300;7.900]**